src/Chapter3.hs

```diff
@@ -970,6 +1067,19 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Week = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Week -> Bool
+isWeekend day = day == Saturday || day == Sunday
+
+nextDay :: Week -> Week
+nextDay Sunday = minBound
```

Comment:
That is right!
You could also use the instance of `Bounded` in the following way in here:

```haskell
nextDay day = if day == maxBound then minBound else succ day
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂